tiledb/sm/query/deletes_and_updates/test/unit_update_condition.cc

```diff
@@ -0,0 +1,127 @@
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+ * THE SOFTWARE.
+ *
+ * @section DESCRIPTION
+ *
+ * Tests the update condition serialization.
+ */
+
+#include "test/src/ast_helpers.h"
+#include "tiledb/sm/query/deletes_and_updates/serialization.h"
+#include "tiledb/sm/storage_manager/context.h"
+
+#include <test/support/tdb_catch.h>
+
+using namespace tiledb::sm;
+using namespace tiledb::sm::deletes_and_updates::serialization;
+
+/**
+ * Make sure a condition and update values are the same after going through
+ * serialization/deserialization.
+ *
+ * @param query_condition Condition to check.
+ * @param update_values Update values to check.
+ */
+void serialize_deserialize_check(
+    QueryCondition& query_condition, std::vector<UpdateValue>& update_values) {
+  auto serialized =
+      serialize_update_condition_and_values(query_condition, update_values);
+  auto&& [deserialized_condition, deserialized_update_values] =
+      deserialize_update_condition_and_values(
+          0, "", serialized.data(), serialized.size());
+
+  CHECK(tiledb::test::ast_equal(
+      query_condition.ast(), deserialized_condition.ast()));
+
+  CHECK(update_values.size() == deserialized_update_values.size());
+
+  for (uint64_t i = 0; i < update_values.size(); i++) {
+    CHECK(
+        update_values[i].field_name() ==
+        deserialized_update_values[i].field_name());
+    CHECK(
+        update_values[i].view().size() ==
+        deserialized_update_values[i].view().size());
+    CHECK(
+        0 == memcmp(
+                 update_values[i].view().content(),
+                 deserialized_update_values[i].view().content(),
+                 update_values[i].view().size()));
+  }
+}
+
+TEST_CASE(
+    "UpdateCondition: Test single value", "[updatecondition][single-value]") {
+  std::string field_name = "foo";
+  char value[] = "bar";
+
+  QueryCondition query_condition;
+  REQUIRE(query_condition
+              .init(
+                  std::string(field_name),
+                  &value,
+                  strlen(value),
+                  QueryConditionOp::LT)
+              .ok());
+
+  std::vector<UpdateValue> values;
+  std::vector<uint8_t> val = {1, 2, 3};
+  values.emplace_back(field_name, val.data(), val.size());
+  serialize_deserialize_check(query_condition, values);
+}
+
+TEST_CASE(
+    "UpdateCondition: Test multiple values",
+    "[updatecondition][multiple-value]") {
+  std::string field_name = "foo";
+  char value[] = "bar";
+
+  QueryCondition query_condition;
+  REQUIRE(query_condition
+              .init(
+                  std::string(field_name),
+                  &value,
+                  strlen(value),
+                  QueryConditionOp::LT)
+              .ok());
+
+  std::vector<UpdateValue> values;
+  std::string field_name1 = "foo";
```

Comment:
Does it make sense to test for corner cases like empty field name, empty value, empty values vector etc? Also to not include the field_name of the query condition in the update values (if I understand well it's not necessary?)

Reply:
Update values will get validated coming in from the client. We do not need the field name of the query condition in the update value as they are separate. Updates can be for example:
UPDATE where w == 5 AND x == 1 set y=2, z = 3.